corehq/apps/ota/views.py

```diff
@@ -263,6 +263,14 @@ def get_restore_response(domain, couch_user, app_id=None, since=None, version='1
     )
 
     app = get_app_cached(domain, app_id) if app_id else None
+
+    #For ICDS only allow users who have access the specific app
+    if settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS:
+        role = couch_user.get_role(domain)
```

Comment:
You would need to catch `DomainMembershipError` here.

Reply:
Ohh! Thanks for pointing out.
Should access be denied to the users who don't have a role?

Reply:
Yes.